dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ResourcePolicyRestRepository.java

```diff
@@ -52,9 +95,249 @@ public ResourcePolicyRest findOne(Context context, Integer id) {
-
     @Override
     public Class<ResourcePolicyRest> getDomainClass() {
         return ResourcePolicyRest.class;
     }
-}
+
+    /**
+     * Find the resource policies matching the uuid of the resource object and/or the specified action
+     * 
+     * @param resourceUuid
+     *            mandatory, the uuid of the resource object of the policy
+     * @param action
+     *            optional, limit the returned policies to the specified action
+     * @param pageable
+     *            contains the pagination information
+     * @return a Page of ResourcePolicyRest instances matching the uuid of the resource object and/or the specified
+     *         action
+     */
+    @PreAuthorize("hasPermission(#resourceUuid, 'dspaceObject', 'ADMIN')")
+    @SearchRestMethod(name = "resource")
+    public Page<ResourcePolicyRest> findByResource(@Parameter(value = "uuid", required = true) UUID resourceUuid,
+                                      @Parameter(value = "action", required = false) String action, Pageable pageable) {
+
+        List<ResourcePolicy> resourcePolisies = null;
+        int total = 0;
+        try {
+            Context context = obtainContext();
+            if (action != null) {
+                int actionId = Constants.getActionID(action);
+                resourcePolisies = resourcePolicyService.findByResouceUuidAndActionId(context, resourceUuid, actionId,
+                        Math.toIntExact(pageable.getOffset()),
+                        Math.toIntExact(pageable.getOffset() + pageable.getPageSize()));
+                total = resourcePolicyService.countByResouceUuidAndActionId(context, resourceUuid, actionId);
+            } else {
+                resourcePolisies = resourcePolicyService.findByResouceUuid(context, resourceUuid,
+                        Math.toIntExact(pageable.getOffset()),
+                        Math.toIntExact(pageable.getOffset() + pageable.getPageSize()));
+                total = resourcePolicyService.countByResourceUuid(context, resourceUuid);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return converter.toRestPage(resourcePolisies, pageable, total, utils.obtainProjection(true));
+    }
+
+    /**
+     * Find the resource policies matching uuid of the eperson and/or the one specified resource object
+     * 
+     * @param epersonUuid
+     *            mandatory, the uuid of the eperson that benefit of the policy
+     * @param resourceUuid
+     *            optional, limit the returned policies to the ones related to the specified resource
+     * @param pageable
+     *            contains the pagination information
+     *
+     * @return It returns the list of explicit matching resource policies, no inherited or broader resource policies
+     *         will be included in the list nor policies derived by groups' membership
+     */
+    @PreAuthorize("hasPermission(#epersonUuid, 'EPERSON', 'READ')")
```

Comment:
Same here, I think we should require WRITE or ADMIN permissions here.  READ may not be tight enough.

Reply:
as above, here the test to verify that user can only access policies that belong to themself
https://github.com/DSpace/DSpace/pull/2604/files#diff-3a5daad2fc92c374f068f92a31c36ec7R393